### PR TITLE
#163400107 Ft ux login 

### PR DESCRIPTION
**Pull request by AlvinMugambi**
**What does this PR do?**
Adds the login page javascript source code for the login function

**Tasks to be completed**
- add the login function that handles the login page
- add relevant IDs to the login input fields

**Relevant PT story**
[#163400107](https://www.pivotaltracker.com/story/show/163400107)

### CODE REVIEW

File: UI/static/login.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  })
+  .then((res) => res.json())
+  .then((data ) => {
+    if (data.status === 200){
+      localStorage.setItem('token', data.token);
+      window.location.href = "../templates/homepage.html";
+    } else{
+    document.getElementById('error').innerHTML = data.error;
+    }
+    console.log(data);
```
Comment (ronaldndirangu) [inline UI/static/login.js]: Please remove the console log statements.

File: UI/static/login.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  })
+
+  .catch(error => {
+    console.error('Error:', error);
```
Comment (ronaldndirangu) [inline UI/static/login.js]: Here you should throw the error or return a descriptive error message. You should not console log it.

**Review by ronaldndirangu — APPROVED** [review]

File: UI/static/login.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+var loginbutton = document.getElementById('loginbutton').addEventListener('click', login)
+
+function login(event){
+  event.preventDefault();
+  let url = 'http://127.0.0.1:5000/api/v2/auth/login';
```
Comment (ronaldndirangu) [inline UI/static/login.js]: Use the heroku url here instead of the local host one

**Review by ronaldndirangu — APPROVED** [review]

File: UI/static/login.js (hunk 1 above)
Comment (AlvinMugambi) [reply UI/static/login.js]: Noted.

File: UI/static/login.js (hunk 3 above)
Comment (AlvinMugambi) [reply UI/static/login.js]: Noted.
Implemented

File: UI/static/login.js (hunk 2 above)
Comment (AlvinMugambi) [reply UI/static/login.js]: On it.
Implemented

**Pull request merged**